frontend/packages/dev-console/src/components/topology2/components/nodes/WorkloadNode.tsx

```diff
@@ -0,0 +1,136 @@
+  const repoIcon = routeDecoratorIcon(workloadData.editUrl, decoratorRadius);
+  const cx = width / 2;
+  const cy = height / 2;
+
+  return (
+    <g>
+      <BaseNode
+        outerRadius={radius}
+        innerRadius={donutStatus && donutStatus.isRollingOut ? radius * 0.45 : radius * 0.55}
+        icon={workloadData.builderImage}
+        kind={workloadData.kind}
+        element={element}
+        {...rest}
+        attachments={[
+          repoIcon && (
+            <Tooltip key="edit" content="Edit Source Code" position={TooltipPosition.right}>
+              <Decorator
+                x={cx + radius - decoratorRadius * 0.7}
+                y={cy + radius - decoratorRadius * 0.7}
+                radius={decoratorRadius}
+                href={workloadData.editUrl}
+                external
+              >
+                <g transform={`translate(-${decoratorRadius / 2}, -${decoratorRadius / 2})`}>
+                  {repoIcon}
+                </g>
+              </Decorator>
+            </Tooltip>
+          ),
+          workloadData.url && (
+            <Tooltip key="route" content="Open URL" position={TooltipPosition.right}>
+              <Decorator
+                x={cx + radius - decoratorRadius * 0.7}
+                y={cy + -radius + decoratorRadius * 0.7}
+                radius={decoratorRadius}
+                href={workloadData.url}
+                external
+                circleRef={urlAnchorRef}
+              >
+                <g transform={`translate(-${decoratorRadius / 2}, -${decoratorRadius / 2})`}>
+                  <ExternalLinkAltIcon style={{ fontSize: decoratorRadius }} alt="Open URL" />
+                </g>
+              </Decorator>
+            </Tooltip>
+          ),
+          build && (
+            <Tooltip
+              key="build"
+              content={`${build.metadata.name} ${build.status && build.status.phase}`}
+              position={TooltipPosition.left}
+            >
+              <Link
+                to={`${resourcePathFromModel(
+                  BuildModel,
+                  build.metadata.name,
+                  build.metadata.namespace,
+                )}/logs`}
+                className="odc2-decorator__link"
+              >
+                <Decorator
+                  x={cx - radius + decoratorRadius * 0.7}
+                  y={cy + radius - decoratorRadius * 0.7}
+                  radius={decoratorRadius}
+                >
+                  <g transform={`translate(-${decoratorRadius / 2}, -${decoratorRadius / 2})`}>
+                    <foreignObject
+                      width={decoratorRadius}
+                      height={decoratorRadius}
+                      style={{ fontSize: decoratorRadius }}
+                    >
+                      <Status status={build.status.phase} iconOnly noTooltip />
+                    </foreignObject>
+                  </g>
+                </Decorator>
+              </Link>
+            </Tooltip>
```

Comment:
There was work done here to add pipeline decorator as well - if I am reading this correctly, this is lost in this change?

Reply:
Good catch. Yes that was recently added and not brought over. We'll have to add that into the new node.
cc @jeff-phillips-18